src/lib/network_pool/snark_pool.ml

```diff
@@ -330,15 +330,21 @@ module Make (Transition_frontier : Transition_frontier_intf) :
                       work ) ] ;
           `Statement_not_referenced )
 
+      (* At this point, verification already has happened *)
       let verify_and_act t ~work ~sender =
         let statements, priced_proof = work in
-        let open Deferred.Or_error.Let_syntax in
         let {Priced_proof.proof= proofs; fee= {prover; fee}} = priced_proof in
         let trust_record =
           Trust_system.record_envelope_sender t.config.trust_system t.logger
             sender
         in
         let log_and_punish ?(punish = true) statement e =
+          (* TODO: For now, we must not punish since we batch across messages received from
+             different senders and we don't isolate the bad proof in a batch, so we cannot
+             properly attribute blame.
+          *)
```

Comment:
make an issue for this?

Reply:
https://github.com/CodaProtocol/coda/issues/5595